### ISSUE TITLE
Partners Portal zip field validation issues

### ISSUE THREAD

**Issue by emilyjablonski** [bloom-housing/bloom#2607]
**Describe the bug**
Partners portal listing form zip code fields are currently accepting alphabetical characters.

**Expected behavior**
The field should only accept numerical characters and be limited to 5 characters. This should be both a backend and a frontend validation. 

**Screenshots**
![Screen Shot 2022-08-08 at 6.11.55 PM.png](https://images.zenhubusercontent.com/601c7ede5322cf8ac29d50c4/03abcdd4-86b2-4a6b-800e-43e851e447c9)

**Comment by ludtkemorgan** [comment]
@emilyjablonski I did a check in the hba prod db and there are a couple of listings that have invalid zip codes. The DB address table is also the same one used for applicants which has a bunch of non-five digit zip codes. Such as 9 digits zip-codes (12345-1234) and random text. So we would not be able to update the DB column to something other than `text` for now. 

I could make the front-end only accept numbers though, especially if we just target the partner site. Should we proceed with that for now?
